File: civitasJAVA/src/civitas/SorpresaPagarCobrar.java
package civitas;

import java.util.ArrayList;

/**
 * @author Jesús Navarro Merino
 * @author José Alberto Gómez García
 */

public class SorpresaPagarCobrar extends Sorpresa {

    private int valor;

    public SorpresaPagarCobrar(String texto, int _valor) {
        super(texto);
        valor=_valor;
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual,todos)){
            informe(actual,todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }
    
    @Override
    protected void informe (int actual, ArrayList<Jugador> todos){
        super.informe(actual, todos);
    }
    
} //Fin de clase SorpresaPagarCobrar

File: civitasRUBY/lib/sorpresa_pagar_cobrar.rb
# encoding:utf-8

module Civitas
  class SorpresaPagarCobrar < Sorpresa
    
    attr_reader :texto, :valor
    
    def initialize(texto,_valor)
      super(texto)
      @valor=_valor
    end
    
    def self.constructorSorpresaPagarCobrar(texto,valor)
      new(texto,valor)
    end
    
    def aplicarAJugador(actual,todos)
      if(jugadorCorrecto(actual, todos))
        informe(actual, todos)
        todos[actual].modificarSaldo(@valor)
      end
    end
    
    def informe(actual,todos)
      super(actual,todos)
    end
    
    public_class_method :new
    
  end
end
